hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUpdateNftsHandlerTest.java

```diff
@@ -219,73 +246,116 @@ void doesntFailWhenMetadataIsEmpty() {
+        when(preHandleContext.body()).thenReturn(transactionBody);
+        when(transactionBody.tokenUpdateNftsOrThrow()).thenReturn(tokenUpdateNftsTransactionBody);
+        when(tokenUpdateNftsTransactionBody.tokenOrElse(TokenID.DEFAULT)).thenReturn(nonFungibleTokenId);
+        when(tokenUpdateNftsTransactionBody.serialNumbers()).thenReturn(List.of(1L));
+        when(preHandleContext.createStore(ReadableTokenStore.class)).thenReturn(readableTokenStore);
+        when(preHandleContext.createStore(ReadableNftStore.class)).thenReturn(readableNftStore);
+        when(readableTokenStore.get(nonFungibleTokenId)).thenReturn(token);
+        when(readableNftStore.get(nonFungibleTokenId, 1L)).thenReturn(nft);
+        when(token.tokenIdOrThrow()).thenReturn(nonFungibleTokenId);
+        when(nft.ownerId()).thenReturn(AccountID.newBuilder().accountNum(1).build());
+        when(token.hasMetadataKey()).thenReturn(false);
+
+        Assertions.assertThatThrownBy(() -> subject.preHandle(preHandleContext))
+                .isInstanceOf(PreCheckException.class)
+                .has(responseCode(TOKEN_HAS_NO_METADATA_KEY));
     }
 
     @Test
-    void nftSerialNotFound() {
-        final List<Long> serialNumbers = new ArrayList<>(Arrays.asList(1L, 2L));
-        readableTokenStore = newReadableStoreWithTokens(
-                Token.newBuilder().tokenId(TOKEN_123).metadataKey(metadataKey).build());
-        writableTokenStore = newWritableStoreWithTokens(Token.newBuilder()
-                .tokenId(TOKEN_123)
-                .metadataKey(metadataKey)
-                .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
-                .treasuryAccountId(ACCOUNT_1339)
-                .supplyKey(TOKEN_SUPPLY_KT.asPbjKey())
-                .totalSupply(10)
-                .build());
-        writableTokenRelStore = newWritableStoreWithTokenRels(TokenRelation.newBuilder()
-                .accountId(ACCOUNT_1339)
-                .tokenId(TOKEN_123)
-                .balance(10)
-                .build());
-        writableNftStore = new WritableNftStore(
-                new MapWritableStates(
-                        Map.of("NFTS", MapWritableKVState.builder("NFTS").build())),
-                CONFIGURATION,
-                storeMetricsService);
+    void preHandle_WhenTokenHasMetadataKeyAndSupplyKey() {
+        when(preHandleContext.body()).thenReturn(transactionBody);
+        when(transactionBody.tokenUpdateNftsOrThrow()).thenReturn(tokenUpdateNftsTransactionBody);
+        when(tokenUpdateNftsTransactionBody.tokenOrElse(TokenID.DEFAULT)).thenReturn(nonFungibleTokenId);
+        when(preHandleContext.createStore(ReadableTokenStore.class)).thenReturn(readableTokenStore);
+        when(readableTokenStore.get(nonFungibleTokenId)).thenReturn(token);
+        when(token.hasMetadataKey()).thenReturn(true);
+        when(token.hasSupplyKey()).thenReturn(true);
+
+        assertThatCode(() -> subject.preHandle(preHandleContext)).doesNotThrowAnyException();
```

Comment:
In addition to nothing "going wrong" it seems like we would also want to verify one of the keys was added as a requirement

Reply:
Added additional verify checks to check the expected invocation of the key to be required.